src/AaveHelper.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {IAavePool} from "./vendored/IAavePool.sol";
+import {IERC20} from "./vendored/IERC20.sol";
+import {SafeERC20} from "./vendored/SafeERC20.sol";
+
+/// @title AaveHelper
+/// @author CoW DAO developers
+/// @notice A helpful helper.
+contract AaveHelper {
```

Comment:
One nit regarding `AaveHelper` contract, is how was called in the diagrams, but probably `AaveCollateralSwap` is a better name. 

I could see many other helpers for other cases `AavePayback` (to pay back a loan for the amount deposited in the helper), `AAveDebtSwap` (to payback a debt and acquire a new one for a new token), etc.